api/desecapi/views.py

```diff
@@ -517,21 +517,23 @@ def get(self, request, *_):
         if domain is None:
             raise NotFound('nohost')
 
-        data = {'A': self._find_ip_v4(request), 'AAAA': self._find_ip_v6(request)}
+        ipv4 = self._find_ip_v4(request)
+        ipv6 = self._find_ip_v6(request)
+
+        data = [
+            {'type': 'A', 'subname': '', 'ttl': 60, 'records': [ipv4] if ipv4 else []},
+            {'type': 'AAAA', 'subname': '', 'ttl': 60, 'records': [ipv6] if ipv6 else []},
+        ]  # FIXME why reset TTL to 60?
 
         with PDNSChangeTracker():
-            for type_, content in data.items():
-                if content is None:
-                    # FIXME encapsulate to RRset.objects.delete_or_pass()?
-                    try:
-                        RRset.objects.get(type=type_, subname='', domain=domain).delete()
-                    except RRset.DoesNotExist:
-                        pass
-                else:
-                    rrset, _ = RRset.objects.get_or_create(defaults={'ttl': 60}, type=type_, subname='', domain=domain)
-                    rrset.ttl = 60  # FIXME why reset TTL to 60?
-                    rrset.save()
-                    RRsetSerializer.set_all_record_contents(rrset, {content})
+            instances = domain.rrset_set.filter(subname='', type__in=['A', 'AAAA']).all()
+            serializer = RRsetSerializer(instances, data=data, many=True, partial=False)
+            serializer.child.fields['records'].validators = []  # FIXME remove this hack after deciding if we allow empty unknowns
```

Comment:
There we have the use case ;-) IMO empty unknowns should be allowed if `PATCH`ing. Am I understanding correctly that in this case, this corresponds to using `partial=False` in the preceding line? If so, we do not have to touch the `validators` attribute and `partial=False` will ensure that `reocrds=[]` is ok.